Resolve weather station id's and create a mapping table

alexhyunminlee opened the pull request:
## Summary

This PR includes code that resolves the weather station id for each bldg_id. This is done by downloding the hpxml file for the bldg_id, extracting out the weather station id, and creating a mapping metadata table for each bldg_id to its corresponding weather station id.

## Implementation

A new script is added inside `utils` called `resolve_weather_station_id.py` that takes care of this.

Closes #48

jpvelez commented on utils/resolve_weather_station_id.py:
```diff
@@ -0,0 +1,759 @@
+
+if __name__ == "__main__":
+    start_time = time.time()
+    product = "resstock"
```
This is a common pattern I've seen you do. Why not grab these fields from the releases json?

alexhyunminlee replied:
But also how would you grab the fields from the releases json? Like would you specify the release name (e.g. `res_2024_tmy3_2` and then the rest (i.e. `product`, `release_year`, `weather_file`, and `release_version`) are filled out automatically?

jpvelez commented on utils/resolve_weather_station_id.py:
```diff
@@ -0,0 +1,759 @@
+    os.makedirs(output_dir, exist_ok=True)
+
+    # Remove duplicates and sort by building ID for better organization
+    if "bldg_id" in df.columns:
```
Is `bldg_id` always supposed to be here? If so, use an `assert`, not a conditional.

jpvelez replied:
Also, this whole section should be a function.

jpvelez commented on utils/resolve_weather_station_id.py:
```diff
@@ -0,0 +1,759 @@
+        # Remove duplicates based on bldg_id, keeping the first occurrence
+        df = df.unique(subset=["bldg_id"], maintain_order=False)
+        final_count = len(df)
+        if original_count != final_count:
```
When would this be the case?

jpvelez commented on utils/resolve_weather_station_id.py:
```diff
@@ -0,0 +1,759 @@
+        final_count = len(df)
+        if original_count != final_count:
+            print(f"Removed {original_count - final_count} duplicate bldg_id entries")
+        df = df.sort("bldg_id")
```
Also, why are we sorting?

alexhyunminlee commented:
@jpvelez Addressed the comments above. One quick note about removing duplicate bldg_id part: I'm not sure why, but fetch_bldg_id keeps returning dulicate bldg_id's. I'll have to do some more troubleshooting to figure why it's doing that, but for now I'm just removing duplicate bldg_id's in this script. I fear that there might be something wrong with how we're building the metadata tables in the first place. I'll have to look into it more to confirm.

alxsmith commented:
ltgm just address the lingering comments from juan-pablo

alxsmith reviewed (changes requested):
address remaining comments then merge
